src/nasuta-sebastian/Fsd.Sebastian.Ng/Fsd.Sebastian.Ng.Ex1/restaurant/src/app/features/products-menu/menu-manger.ts

```diff
@@ -0,0 +1,48 @@
+import { SelectedListItem } from '../models/selected-list-item';
+import { Product } from '../models/product';
+
+export class MenuManger {
+    selectedProducts: SelectedListItem[];
+    totalPrice: number;
+
+    constructor(items: SelectedListItem[]) {
+        this.selectedProducts = items;
+    }
+
+    addProduct(product: Product) {
+        const select = this.getSelectedProduct(product);
+
+        if (select == null) {
+            this.selectedProducts.push(new SelectedListItem(product, 1));
+        } else {
+            select.quantity++;
+        }
+
+        this.recalculateBill();
+    }
+
+    removeProduct(product: Product) {
+        const select = this.getSelectedProduct(product);
+
+        select.quantity--;
+
+        if (select.quantity === 0) {
+            this.selectedProducts.splice(this.selectedProducts.indexOf(select), 1);
+        }
+
+        this.recalculateBill();
+    }
+
+    getSelectedProduct(product: Product) {
+        return this.selectedProducts.find(item => item.product.name === product.name);
+    }
+
+    recalculateBill() {
+        let sum = 0;
+        for (let i = 0; i < this.selectedProducts.length; i++) {
+            sum = sum + this.selectedProducts[i].product.price * this.selectedProducts[i].quantity;
```

Comment:
pętla jest ok, ale dobrze wygląda użycie metody forEach dla obiektu tablicy np. this.selectedProducts.forEach(p=> ... )